bundles/org.openhab.binding.panamaxfurman/src/main/java/org/openhab/binding/panamaxfurman/internal/transport/PanmaxFurmanConnector.java

```diff
@@ -0,0 +1,157 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Wraps the communication to a Panamax/Furman Power Conditioner with a BlueBOLT-CV1 or BlueBOLT-CV2 card via telnet
+ *
+ * @author Dave Badia - Initial contribution
+ */
+@NonNullByDefault
+public class PanmaxFurmanConnector {
+    private final Logger logger = LoggerFactory.getLogger(PanmaxFurmanConnector.class);
+
+    private static final int CONNECT_TIMEOUT_MILLIS = 10_000;
+
+    private final String connectionName;
+    private final String address;
+    private final int port;
+    private final int socketReadTimeoutSeconds;
+    private final PanamaxFurmanCommunicationEventListener eventGenerator;
+
+    private final Semaphore readerReadySemaphore = new Semaphore(1);
+    private final Object ioMutex = new Object();
+
+    /**
+     * socketConnection will be recreated as necessary
+     */
+    private SocketConnection socketConnection;
+
+    public PanmaxFurmanConnector(String connectionName, String address, int port, int socketReadTimeoutSeconds,
+            PanamaxFurmanCommunicationEventListener eventGenerator) throws PanamaxFurmanDeviceUnavailableException {
+        this.connectionName = connectionName;
+        this.address = address;
+        this.port = port;
+        this.socketReadTimeoutSeconds = socketReadTimeoutSeconds;
+        this.eventGenerator = eventGenerator;
+        this.socketConnection = connectToDevice();
+    }
+
+    public void sendRequestToDevice(@Nullable String textToSend) {
+        sendRequestToDevice(textToSend, true);
+    }
+
+    public synchronized void sendRequestToDevice(@Nullable String command, boolean retry) {
+        if (command == null) {
+            return;
+        }
+        synchronized (ioMutex) {
+            try {
+                socketConnection.writer.write(command + "\r");
+                socketConnection.writer.flush();
+                logger.debug(">> {}    @{}", command, connectionName);
+            } catch (IOException e) {
+                logger.debug("Connection with device is broken. retry={}  @{}", retry, connectionName, e);
+                if (retry) {
+                    try {
+                        this.socketConnection = connectToDevice();
+                        sendRequestToDevice(command, false);
+                    } catch (PanamaxFurmanDeviceUnavailableException e1) {
+                        eventGenerator.onConnectivityEvent(new PanamaxFurmanConnectionBrokenEvent(e1));
+                        // Nothing to log, error was logged in connectToDevice()
+                        // Don't rethrow as not being able to connect sometimes is a normal condition
+                    }
+                }
+            }
+        }
+    }
+
+    private synchronized SocketConnection connectToDevice() throws PanamaxFurmanDeviceUnavailableException {
+        synchronized (ioMutex) {
+            if (socketConnection != null && !socketConnection.inputReaderThread.isReading()) {
```

Comment:
socketConnection is `@NonNull` so there should not be a need to check for null, or should it be `@Nullable`